<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1271</width>
    <height>887</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="1">
     <widget class="CameraStreamWidget" name="cameraStreamWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>720</width>
        <height>270</height>
       </size>
      </property>
      <property name="toolTip">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="dashboardGroupBox">
      <property name="title">
       <string>Dashboard</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QCheckBox" name="completedMotorCalibrationCheckBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Motor calibration</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="completedMotorCameraCalibrationCheckBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Motor-Camera calibration</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_13">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Positioning error:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_11">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Encoder error:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_9">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Projection error:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_14">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>0.00 px</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="completedCameraCalibrationCheckBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Camera calibration</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLabel" name="acquisitionFPSLabel">
         <property name="text">
          <string>0.0 Hz</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_12">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>0 ticks</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="projectionErrorLabel">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>0.00 px</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Acquisition FPS:</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="testTab">
       <attribute name="title">
        <string>Test</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="2" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Azimuth</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="altValueSpinBox">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButton_4">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Record Video</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_3">
            <property name="text">
             <string>Take Picture</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="aziValueSpinBox">
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="stopCameraCaptureThreadButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="startCameraCaptureThreadButton">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Altitude</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QPushButton" name="sendMotorPositionButton">
            <property name="text">
             <string>Send Position</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Show Processing</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QRadioButton" name="showProcessingNoneRadioButton">
             <property name="text">
              <string>None</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">showProcessingButtonGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="showProcessingThresholdingRadioButton">
             <property name="text">
              <string>LED thresholding</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">showProcessingButtonGroup</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="showProcessingBGSubRadioButton">
             <property name="text">
              <string>Background subtraction</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">showProcessingButtonGroup</string>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="cameraCalibrationTab">
       <attribute name="title">
        <string>Camera Calibration</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5" stretch="0,0,0">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string># Camera Calibration
This will define the projection matrices of the cameras. Load matrices from a saved file or create new calibration.
For a new calibration take 20-30 images of the chessboard, each time in a different position.</string>
          </property>
          <property name="textFormat">
           <enum>Qt::TextFormat::MarkdownText</enum>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QRadioButton" name="selectCal3DFromSavedFile">
            <property name="text">
             <string/>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">selectCal3DTypeButtonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="loadCal3DFromSavedFileGroupBox">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>469</height>
             </size>
            </property>
            <property name="title">
             <string>Load From Saved File</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>File:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="calibrateCal3DFromLoadedFile">
               <property name="text">
                <string>Calibrate</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="savedCal3DFilesRefreshButton">
               <property name="text">
                <string>Refresh</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="savedCal3DFilesComboBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QRadioButton" name="selectNewCal3D">
            <property name="text">
             <string/>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">selectCal3DTypeButtonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="newCal3DGroupBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="title">
             <string>New Calibration</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="6" column="1">
              <widget class="QPushButton" name="newCal3DSaveToFileButton">
               <property name="text">
                <string>Save to File</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Name:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="chessColumnsSpinBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>9</number>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QPushButton" name="calibrateNewCal3DButton">
               <property name="text">
                <string>Calibrate</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Saved Image Pairs:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="chessRowsSpinBox">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="value">
                <number>6</number>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="newCal3DStartImageCaptureButton">
               <property name="text">
                <string>Start Image Capture</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QPushButton" name="newCal3DSaveImagePairButton">
               <property name="text">
                <string>Save Image Pair</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QPushButton" name="newCal3DStopImageCaptureButton">
               <property name="text">
                <string>Stop Image Capture</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Chessboard Rows</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="numberOfSavedImagePairsLabel">
               <property name="text">
                <string>0</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>Chessboard Columns</string>
               </property>
              </widget>
             </item>
             <item row="6" column="2">
              <widget class="QPushButton" name="cancelNewCal3DButton">
               <property name="text">
                <string>Cancel</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QDoubleSpinBox" name="squareSideLengthSpinBox">
               <property name="minimum">
                <double>1.000000000000000</double>
               </property>
               <property name="value">
                <double>20.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Square Side Length (mm)</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="newCal3DNameLineEdit">
               <property name="placeholderText">
                <string>Enter new calibration name</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="motorCameraCalibrationTab">
       <attribute name="title">
        <string>Motor-Camera Calibration</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="2" column="1">
         <widget class="QLabel" name="label_18">
          <property name="text">
           <string>(0, 0)</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="thresholdSpinBox">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>230</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>Threshold</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="startMotorCameraCalibrationButton">
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string>Motor Positions (Azi, Alt)</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="stopMotorCameraCalibrationButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gameTab">
       <attribute name="title">
        <string>Game</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1271</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="enabled">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CameraStreamWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>CameraStreamWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="selectCal3DTypeButtonGroup"/>
  <buttongroup name="showProcessingButtonGroup"/>
 </buttongroups>
</ui>
